### PR TITLE
Feature/Android-Ponto

### PR DESCRIPTION
**Pull request by ArturKlajnerok**

### CODE REVIEW

File: android/ponto-sample/src/com/wikia/ponto/sample/modules/Messaging.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                body = article.optString("body");
+            }
+        } catch (JSONException e) {
+            Log.e(TAG, "JSONException while parsing messaging data", e);
+        }
+
+        Toast.makeText(mContext, body, Toast.LENGTH_SHORT).show();
+    }
+
+    public void getSomeData() {
```
Comment (aniaMu) [inline android/ponto-sample/src/com/wikia/ponto/sample/modules/Messaging.java:58]: empty method

File: android/ponto/src/com/wikia/ponto/Ponto.java [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import android.util.Log;
+import android.webkit.JavascriptInterface;
+import android.webkit.WebView;
+
+/**
+ * TODO description
```
Comment (aniaMu) [inline android/ponto/src/com/wikia/ponto/Ponto.java]: Please add description for this class

File: android/ponto/src/com/wikia/ponto/Ponto.java [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+    public Ponto(WebView webView, String classPackage) {
+        mCallbacks = new HashMap<String, RequestCallback>();
+        mClassPackage = classPackage;
+        mWebView = webView;
+        mWebView.getSettings().setJavaScriptEnabled(true);
+        mWebView.addJavascriptInterface(new PontoProtocol(), PontoProtocol.TAG);
+    }
+
+    /**
+     * TODO description
```
Comment (aniaMu) [inline android/ponto/src/com/wikia/ponto/Ponto.java]: please add description for this method

File: android/ponto/src/com/wikia/ponto/Ponto.java [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+            int responseType = RESPONSE_ERROR;
+            JSONObject responseParams = new JSONObject();
+            try {
+                Class<?> cls = Class.forName(mClassPackage + className);
+                Constructor<?> constructor = cls.getConstructor(android.content.Context.class);
+                Object object = constructor.newInstance(mWebView.getContext());
+                if (params != null && !params.equalsIgnoreCase(NULL_STRING)) {
+                    Method method = cls.getDeclaredMethod(methodName, String.class);
+                    method.invoke (object, params);
```
Comment (aniaMu) [inline android/ponto/src/com/wikia/ponto/Ponto.java]: formatting

File: android/ponto/src/com/wikia/ponto/Ponto.java [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+         * @param async 
+         */
+        @JavascriptInterface
+        public void request(String execContext, String className, String methodName, String params, 
+                String callbackId, String async) {
+
+            int responseType = RESPONSE_ERROR;
+            JSONObject responseParams = new JSONObject();
+            try {
+                Class<?> cls = Class.forName(mClassPackage + className);
```
Comment (aniaMu) [inline android/ponto/src/com/wikia/ponto/Ponto.java:128]: is there a guarantee of a proper format of package.className? Will the dot between always be there?

File: android/ponto/src/com/wikia/ponto/Ponto.java [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+        private int getResponeTypeFromParams(String params) {
+            JSONObject responseParams;
+            int type = RESPONSE_COMPLETE;
+            try {
+                responseParams = new JSONObject(params);
+                if (responseParams != null) {
+                    type = responseParams.optInt("type");
+                }
+            } catch (JSONException e) {
+                Log.e(TAG, "JSONException while parsing messaging data", e);
```
Comment (aniaMu) [inline android/ponto/src/com/wikia/ponto/Ponto.java:221]: response type should be incomplete when JSONException occurs

File: android/ponto/src/com/wikia/ponto/Ponto.java (hunk 5 above)
Comment (ArturKlajnerok) [inline android/ponto/src/com/wikia/ponto/Ponto.java:128]: If package provided will be wrong "ClassNotFoundException" will be cached. I've added the dot in constructor, so only proper package with classes that can be called from WebView need to be provided.

File: android/ponto/src/com/wikia/ponto/Ponto.java (hunk 6 above)
Comment (ArturKlajnerok) [inline android/ponto/src/com/wikia/ponto/Ponto.java:221]: that's correct, fixed

**Comment by aniaMu** [comment]
LGTM :+1:

**Pull request merged**